R/stat-density-2d.r

```diff
@@ -66,6 +71,7 @@ StatDensity2d <- ggproto("StatDensity2d", Stat,
       StatContour$compute_panel(df, scales, bins, binwidth)
     } else {
       names(df) <- c("x", "y", "density", "group")
+      df$scaled <- df$density / max(df$density)
```

Comment:
Thanks for taking the time to consider this addition!

I wasn't entirely happy with the switch between `ndensity` to `scaled` either and would actually prefer to use `ndensity` here. However, in `stat_density` [the 1d version], the rescaled statistic is called `scaled` while in `stat_bin` it's called `ndensity` or `ncount`. 

Since `stat_density_2d` is the 2d version of `stat_density` I tried to mirror the syntax by naming it `scaled`.  I think it might make the most sense to use one syntax throughout all the stat functions (`ndensity`), but I will leave that to your discretion. To summarize, I think there are a three options. 

1. Mirror the syntax from the 1d versions of the stat layers such that `stat_density_2d` and `stat_density` use `scaled` while `stat_bin2d`, `stat_hexbin`, and `stat_bin` use `ndensity` and `ncount`. (implemented this PR)

2. (2a) Switch over to a single syntax for the 2d stats with this version - `ndensity` replaces `scaled` in `stat_density_2d`. 

3. (2b) Switch over to a single syntax for all density based stats -  `ndensity` replaces `scaled` in both `stat_density` and `stat_denisty_2d`. I suppose you would want to just duplicate `scaled` and name the new variable `ndensity` to avoid breaking existing code. 

--
RE: na.rm = TRUE; you're correct, I've added that to my local version for my next PR. I can submit that change once `scaled` vs. `ndensity` is settled and I'll update the documentation and news as well.

Reply:
I think option 2b (uniform naming with duplicated `scaled` column for `stat_density`) is best. But I'd like to hear @hadley's input as well.

Reply:
Yeah, 2b sounds good (assume it doesn't affect existing code, in which case scaled should be added, instead of replacing the existing variable name)